worlds/hades/Client.py

```diff
@@ -0,0 +1,409 @@
+            self.location_name_to_id = self.get_location_name_to_id()
+            if "death_link" in self.hades_slot_data and self.hades_slot_data["death_link"]:
+                asyncio.create_task(self.update_death_link(True))
+                self.deathlink_enabled = True
+            self.is_connected = True
+            self.is_receiving_items_from_connect_package = True 
+
+        if cmd == "RoomInfo":
+            # What should be done when room info is sent.
+            self.seed_name = args["seed_name"]
+        
+        if cmd == "RoomUpdate":
+            if "checked_lodations" in args:
+                collect_locations_cache = ""
+                for location in args["checked_locations"]:
+                    collect_locations_cache += self.location_names.lookup_in_slot(location) + "-"
+                if len(collect_locations_cache) > 0:
+                    collect_locations_cache = collect_locations_cache[:-1]
+                    subsume.Send(styx_scribe_send_prefix + "Locations collected:" + collect_locations_cache)
+
+        if cmd == "ReceivedItems":
+            # What should be done when an Item is recieved.
+            # NOTE THIS GETS ALL ITEMS THAT HAVE BEEN RECIEVED! WE USE THIS FOR RESYNCS!
+            for item in args["items"]:
+                self.cache_items_received_names += [self.item_names.lookup_in_slot(item.item)]
+
+            # We ignore sending the package to hades if just connected,
+            # since the game is not ready for it (and will request it itself later)
+            if self.is_receiving_items_from_connect_package:
+                return
+            self.send_items()
+
+        if cmd == "LocationInfo":
+            if self.creating_location_to_item_mapping:
+                self.creating_location_to_item_mapping = False
+                asyncio.create_task(self.create_location_to_item_dictionary(args["locations"]))
+                return
+            super().on_package(cmd, args)
+            
+        if cmd == "Bounced":
+            if "tags" in args:
+                if "DeathLink" in args["tags"]:
+                    self.on_deathlink(args["data"])
+
+    def send_items(self):
+        payload_message = self.parse_array_to_string(self.cache_items_received_names)
+        subsume.Send(styx_scribe_send_prefix + "Items Updated:" + payload_message)
+
+    def parse_array_to_string(self, array_of_items):
+        message = ""
+        for itemname in array_of_items:
+            message += itemname
+            message += ","
+        return message
+
+    async def send_location_check_to_server(self, message):
+        sendingLocationsId = []
+        sendingLocationsName = message
+        payload_message = []
+        sendingLocationsId += [self.location_name_to_id[sendingLocationsName]]
+        payload_message += [{"cmd": "LocationChecks", "locations": sendingLocationsId}]
+        asyncio.create_task(self.send_msgs(payload_message))
```

Comment:
This whole function can be reduced to one function call. The variables don't need to be initialized to empty arrays before getting concatenated with another new array. The intermediate variables serve no purpose except readability, but you can format `send_msgs` in a pleasing way to avoid them.